File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int[][] res=new int[intervals.length][2];
        Arrays.sort(intervals,(a,b)->Integer.compare(a[0],b[0]));
        int[] newinterval=intervals[0];
        int i=0;
        boolean first=true;
        for(int[] interval:intervals){
            if(interval[0]<=newinterval[1]){
                newinterval[1]=Math.max(interval[1],newinterval[1]);
            }
            else{
                res[i++]=newinterval;
                newinterval=interval;
            }
        }
        res[i++]=newinterval;
        return Arrays.copyOf(res, i);
        
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        if(intervals.size()<=1) return intervals;
        sort(intervals.begin(),intervals.end());
        vector<int> newinterval=intervals[0];
        //res.push_back(newinterval);
        bool first=true;
        for(vector<int> interval:intervals){
            if(first){
                first=false;
                continue;
            }
            if(interval[0]<=newinterval[1]){ //overlapping
            newinterval[1]=max(interval[1],newinterval[1]);
            }
            else{//starting point anotther interval
              res.push_back(newinterval);
              newinterval=interval;
            }
        }
         res.push_back(newinterval); 
        return res;
    }
};